### ISSUE TITLE
Mix randomly generating incomplete mix-manifest.json

### ISSUE THREAD

**Issue by NathanVss** [laravel-mix/laravel-mix#3232]
-   Laravel Mix Version: 6.0.43
-   Node Version (`node -v`): v16.14.0
-   NPM Version (`npm -v`): 8.3.1
-   OS: Debian Buster Docker running in MacOS Monterey ( npm run watch is run inside the Docker )

### Description:

I randomly get the "Unable to locate Mix file: /js/Pages/ScenarioCreatePage.js." error when running `npm run watch` ( Laravel 9 ) 

Sometimes the mix-manifest.json misses the line for "ScenarioCreatePage.js", it seems to be really random when working and running `npm run watch` sometimes the compilation gives a complete manifest like this:

`
{
    "/js/Pages/AuthLoginPage.js": "/js/Pages/AuthLoginPage.js?id=6096dd347f70ff8cd4fb8026d83baf2b",
    "/js/Pages/AuthRegisterPage.js": "/js/Pages/AuthRegisterPage.js?id=13c50a77c598b0c27ac4bce881d18a29",
    "/js/Pages/ScenarioCreatePage.js": "/js/Pages/ScenarioCreatePage.js?id=c3d1aa004e3bde56c82a326bfee6b5a9",
    "/js/Pages/AuthVerifyEmailPage.js": "/js/Pages/AuthVerifyEmailPage.js?id=bd5eeb258a0375f7c5282b1c92f5f5a3",
    "/js/Pages/ScenariosListPage.js": "/js/Pages/ScenariosListPage.js?id=9237d9a9e6f7aa23c65410233368329b",
    "/js/Pages/RunsListPage.js": "/js/Pages/RunsListPage.js?id=4c260eabffd153b14a748d2d042c09f0",
    "/js/index.js": "/js/index.js?id=577c62edcc879306b835f07c2459fd18",
    "/js/manifest.js": "/js/manifest.js?id=1c81f5e66fe755a71bd16fb6bd23cefe",
    "/css/tailwind.css": "/css/tailwind.css?id=63d6a067c3694797e72ce6a66b6cacef",
    "/css/app.css": "/css/app.css?id=7d20650220e261c26634914b147afefa",
    "/js/vendors.js": "/js/vendors.js?id=48371d4d73cae18d174b34121384d896",
    "/css/build.css": "/css/build.css?id=a92fe684993e0ecd9645a7ab91cfa155"
}
`

and sometimes an incomplete one:

`
{
    "/js/Pages/AuthLoginPage.js": "/js/Pages/AuthLoginPage.js?id=6096dd347f70ff8cd4fb8026d83baf2b",
    "/js/Pages/AuthRegisterPage.js": "/js/Pages/AuthRegisterPage.js?id=13c50a77c598b0c27ac4bce881d18a29",
    "/js/Pages/AuthVerifyEmailPage.js": "/js/Pages/AuthVerifyEmailPage.js?id=bd5eeb258a0375f7c5282b1c92f5f5a3",
    "/js/Pages/ScenariosListPage.js": "/js/Pages/ScenariosListPage.js?id=9237d9a9e6f7aa23c65410233368329b",
    "/js/Pages/RunsListPage.js": "/js/Pages/RunsListPage.js?id=4c260eabffd153b14a748d2d042c09f0",
    "/js/index.js": "/js/index.js?id=577c62edcc879306b835f07c2459fd18",
    "/js/manifest.js": "/js/manifest.js?id=1c81f5e66fe755a71bd16fb6bd23cefe",
    "/css/tailwind.css": "/css/tailwind.css?id=63d6a067c3694797e72ce6a66b6cacef",
    "/css/app.css": "/css/app.css?id=7d20650220e261c26634914b147afefa",
    "/js/vendors.js": "/js/vendors.js?id=48371d4d73cae18d174b34121384d896",
    "/css/build.css": "/css/build.css?id=a92fe684993e0ecd9645a7ab91cfa155"
}
`

( You can see that ScenarioCreatePage.js is missing here ) 

Few things
- The first mix-manifest.json generated when starting the watch command is always complete.
- Is is always ScenarioCreatePage that can be missing sometimes, the other files are always present.
- I tried the change the order of the .ts(...) calls in webpack.mix.js but it doesn't affect the bug.
- ScenarioCreatePage.tsx is really small (20 lines), it only contains a ReactDOM.render call, the same goes for all other *Page files, so nothing particular in this file.

### Steps To Reproduce:

Here is my `webpack.mix.js`

`mix
    .ts('resources/src/Auth/LoginPage.tsx', 'public/js/Pages/AuthLoginPage.js')
    .ts('resources/src/Auth/RegisterPage.tsx', 'public/js/Pages/AuthRegisterPage.js')
    .ts('resources/src/Scenarios/Create/ScenarioCreatePage.tsx', 'public/js/Pages/ScenarioCreatePage.js')
    .ts('resources/src/Auth/VerifyEmailPage.tsx', 'public/js/Pages/AuthVerifyEmailPage.js')
    .ts('resources/src/Scenarios/List/ScenariosListPage.tsx', 'public/js/Pages/ScenariosListPage.js')
    .ts('resources/src/Runs/RunsListPage.tsx', 'public/js/Pages/RunsListPage.js')
    .ts('resources/src/index.ts', 'public/js/index.js')
    .react()
    .extract('public/js/vendors.js')
    .postCss('resources/css/tailwind.css', 'public/css/tailwind.css', [
        require('postcss-import'),
        require('tailwindcss'),
        require('autoprefixer'),
    ])
    .sass('resources/sass/app.scss', 'public/css/app.css')
    .combine([
        'public/css/tailwind.css',
        'public/css/app.css'
    ], 'public/css/build.css')
    .version()
    .dump();
`

This bug is really strange and I'm used to work with mix and this never happened to me in the past. The main difference is that I'm trying Laravel 9 on this new project.

Thank you in advance 🙏

**Comment by thecrypticace** [comment]
Can you downgrade to 6.0.42 and see if it still happens for you?

**Comment by thecrypticace** [comment]
I've noticed this morning that `mix.ts()` & manifest generation doesn't work for _production builds_. However downgrading Mix *did not* fix it for me which makes me think some unrelated change has started causing problems. I'd like to just verify my findings before I start diving into what may be the cause here.

**Comment by thecrypticace** [comment]
Yeah, I'm thinking something external to Mix. But kinda hoping we can still fix it anyway. Thanks for the report. I'm gonna try to look into this one this evening (I hope). I've quite quite a handful of things on my list I need to look at but this DST time change is kicking my tail. 🙃
